runtime/shared_heap.c

```diff
@@ -47,6 +47,7 @@ typedef struct pool {
   value* next_obj;
   caml_domain_state* owner;
   sizeclass sz;
+  uintnat p[4];  /* padding to align the beginning of slabs to cache line */
 } pool;
 CAML_STATIC_ASSERT(sizeof(pool) == Bsize_wsize(POOL_HEADER_WSIZE));
```

Comment:
Can you also consider adding a static assert to relate the size of pool to the allocation alignment.

```
CAML_STATIC_ASSERT(sizeof(pool) == Wsize_bsize(Allocation_alignment_bsize));
```

Reply:
Is `Wsize_bsize` needed? We have bytes on both sides. I think LHS will be halved on 32-bit, so we could do the same for RHS. For now, I've added the following, which is a bit weaker: 

`CAML_STATIC_ASSERT(sizeof(pool) % Cache_line_bsize == 0);`

Reply:
Ok. You are right about bytes on both sides.

The new assertion works for me.